### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by Maghoumi**
Update README.md:
* Improve and shorten.
* Include links to blogs and tutorials.
* Remove incorrect info about non-existent branches.
* Add a header and a diagram.
* Add a note about incremental deduplication.

## Description
Make minor revisions to the landing page. Include a link to the TinyStories blog post.

## Checklist
- [X] I am familiar with the [Contributing Guide](https://github.com/NVIDIA/NeMo-Curator/blob/main/CONTRIBUTING.md).
- [X] New or Existing tests cover these changes.
- [X] The documentation is up to date with these changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,54 +1,65 @@
...
+  <a href="https://github.com/NVIDIA/NeMo-Curator/graphs/contributors">![NVIDIA/NeMo-Curator](https://img.shields.io/github/contributors/NVIDIA/NeMo-Curator)</a>
+  <a href="https://github.com/NVIDIA/NeMo-Curator/releases">![https://github.com/NVIDIA/NeMo-Curator/releases](https://img.shields.io/github/release/NVIDIA/NeMo-Curator)</a>
+  <a href="https://pypi.org/project/nemo-curator/">![https://github.com/Naereen/badges/](https://badgen.net/badge/open%20source/❤/blue?icon=github)</a>
 
-At the core of the NeMo Curator is the `DocumentDataset` which serves as the the main dataset class. It acts as a straightforward wrapper around a Dask `DataFrame`. The Python library offers easy-to-use methods for expanding the functionality of your curation pipeline while eliminating scalability concerns.
+</div>
 
-## Key Features
+# NeMo Curator
+🚀 **The GPU-accelerated open source framework for efficient large language models data curation** 🚀
```
Comment (ryantwolf) [inline README.md:12]: 1. Can we capitalize this like a title?
    ```suggestion
    🚀 **The GPU-Accelerated Open Source Framework for Efficient Large Language Models Data Curation** 🚀
    ```
1. Not sure how much of this is for SEO, but I personally would prefer to omit the "Open Source" part. It feels self-evident given that we are on GitHub.
1. I feel like it should be "Large Language Model Data Curation", not "Large Language Models Data Curation". Am I wrong?

File: README.md [hunk 2]
```diff
@@ -1,54 +1,65 @@
...
 
-NeMo Curator provides a collection of scalable data-mining modules. Some of the key features include:
+<p align="center">
+  <img src="./docs/user-guide/images/diagram.png" alt="diagram"/>
```
Comment (ryantwolf) [inline README.md:15]: I am personally not a fan of having a diagram, but after talking with others they have found it useful. I still have a couple of critiques of this diagram in particular.
1. Too much detail for the first thing the user sees. I think a smaller, simpler diagram would serve us better. 
2. Slightly confusing information. There are two "Language Detections" that might confuse the user. We would need to go more in detail to explain why each of them are needed than what is good for a diagram like this.
3. The text is too small.
4. The diagram aspect ratio is too wide.

I'm not sure what the best way to proceed is. We should perhaps have a separate meeting to discuss how to improve it. Do we know of any graphic designers we can reach out to to help refine this? I can make suggestions all day, but actually producing a better version is beyond my skillset.

File: README.md
Comment (ryantwolf) [inline README.md:1]: I know you didn't add this, but at the bottom of this file:
> To contribute an article to the collection, please submit a pull request to the ``gh-pages-src`` branch of this repository. For detailed information, please consult the README located at the [gh-pages-src branch](https://github.com/NVIDIA/NeMo/tree/gh-pages-src#readme).

this should be removed. Not sure why this was added in the first place but there is no `gh-pages-src` branch. This only applies to NeMo, not NeMo Curator.

**Review by ryantwolf — CHANGES_REQUESTED** [review]
I have a couple of changes I'd like to see, but overall nice job!

File: README.md
Comment (Maghoumi) [reply README.md:1]: Done.

File: README.md (hunk 2 above)
Comment (Maghoumi) [reply README.md:15]: Thanks for your comments. Adding @arhamm1 who made the diagram for awareness.
I think it'd be great to have a sync on the diagram with all the stakeholders.

In the meantime, let me know if you want me to remove this diagram from the PR.

File: README.md [hunk 3]
```diff
@@ -1,54 +1,65 @@
...
+  - Flexible interface for extending to your own classifier network
 
-- Identification tools for removing addresses, credit card numbers, social security numbers, and more
+- [Personal identifiable information (PII) redaction](docs/user-guide/personalidentifiableinformationidentificationandremoval.rst) for removing addresses, credit card numbers, social security numbers, and more
 
 These modules offer flexibility and permit reordering, with only a few exceptions. In addition, the [NeMo Framework Launcher](https://github.com/NVIDIA/NeMo-Megatron-Launcher) provides pre-built pipelines that can serve as a foundation for your customization use cases.
 
 ## Resources
 
+- Blog posts
```
[the remark this answers is not in the document]
Comment (Maghoumi) [reply README.md:62]: Moved. Not sure if you noticed but the bullet points were ordered lexicographically 😃

File: README.md [hunk 4]
```diff
@@ -110,18 +119,16 @@ pip install --extra-index-url https://pypi.nvidia.com nemo-curator[cuda12x]
...
-If you want the latest commit inside the container, uninstall the existing version using:
+The latest release of NeMo Curator comes preinstalled in the [NeMo Framework Container](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/nemo/tags). If you want the latest commit inside the container, uninstall the existing version using:
 
 ```bash
 pip uninstall nemo-curator
 ```
 And follow the instructions for installing from source from [above](#from-source).
 
-## Use the Python Library
+### Python API Quick Example
```
[the remark this answers is not in the document]
Comment (Maghoumi) [reply README.md:132]: Agreed. Changed it to "Use NeMo Curator" to stay consistent with the other headers. Our technical writer Jennifer was very particular about the wording of the headers for consistency with the NeMo FW docs.

File: README.md [hunk 5]
```diff
@@ -130,32 +137,35 @@ The following snippet demonstrates how to create a small data curation pipeline
...
+### Explore NeMo Curator Tutorials
 
-To get started with NeMo Curator, you can follow the tutorials available here: [Tutorials]
-(https://github.com/NVIDIA/NeMo-Curator/tree/main/tutorials). These tutorials include:
+To get started with NeMo Curator, you can follow the tutorials [available here](https://github.com/NVIDIA/NeMo-Curator/tree/main/tutorials). These tutorials include:
 
-- `tinystories` which focuses on data curation for training from scratch.
-- `peft-curation` which focuses on data curation for parameter-efficient fine-tuning use-cases.
+- `tinystories` which focuses on data curation for training LLMs from scratch.
+- `peft-curation` which focuses on data curation for LLM parameter-efficient fine-tuning (PEFT) use-cases.
```
[the remark this answers is not in the document]
Comment (Maghoumi) [reply README.md]: Done.

**Review by Maghoumi — COMMENTED** [review]
@ryantwolf Thanks for reviewing. I made all the changes and pushed. Please take a look when you get a chance.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ryantwolf) [reply README.md:62]: Ah I hadn't noticed that. I still think the bottom is the best place though, so thanks for moving it.

File: README.md [hunk 6]
```diff
@@ -130,32 +138,38 @@ The following snippet demonstrates how to create a small data curation pipeline
...
 
-To get started with NeMo Curator, you can follow the tutorials available here: [Tutorials]
-(https://github.com/NVIDIA/NeMo-Curator/tree/main/tutorials). These tutorials include:
+To get started with NeMo Curator, you can follow the tutorials [available here](https://github.com/NVIDIA/NeMo-Curator/tree/main/tutorials). These tutorials include:
 
-- `tinystories` which focuses on data curation for training from scratch.
-- `peft-curation` which focuses on data curation for parameter-efficient fine-tuning use-cases.
+- [`tinystories`](https://github.com/NVIDIA/NeMo-Curator/tree/main/tutorials/tinystories) which focuses on data curation for training LLMs from scratch.
+- [`peft-curation`](https://github.com/NVIDIA/NeMo-Curator/tree/main/tutorials/peft-curation) which focuses on data curation for LLM parameter-efficient fine-tuning (PEFT) use-cases.
+- [`distributed_data_classification`](https://github.com/NVIDIA/NeMo-Curator/tree/main/tutorials/distributed_data_classification) which focuses on using the quality and domain classifiers to help with data annotation and blending.
```
Comment (ryantwolf) [inline README.md:160]: Nit: There is no data blending that occurs in this tutorial.
```suggestion
- [`distributed_data_classification`](https://github.com/NVIDIA/NeMo-Curator/tree/main/tutorials/distributed_data_classification) which focuses on using the quality and domain classifiers to help with data annotation.
```

File: README.md [hunk 7]
```diff
@@ -130,32 +138,38 @@ The following snippet demonstrates how to create a small data curation pipeline
 dataset = download_common_crawl("/datasets/common_crawl/", "2021-04", "2021-10", url_limit=10)
 # Build your pipeline
 curation_pipeline = Sequential([
+  # Fix unicode
   Modify(UnicodeReformatter()),
+  # Discard short records
   ScoreFilter(WordCountFilter(min_words=80)),
+  # Discard low-quality records
   ScoreFilter(FastTextQualityFilter(model_path="model.bin")),
+  # Discard records from irrelevant tasks
```
Comment (ryantwolf) [inline README.md:147]: This is a bit inaccurate. What this is actually doing is ensuring that evaluation metrics don't leak into the training data. Think of it like this: The model shouldn't see/memorize the answers to the final exam. I think
- "Discard records from the evaluation metrics"
- "Prevent test set leakage"
- "Prevent test set contamination"

I'll let you decide on what you think is the best description is.

File: README.md (hunk 2 above)
Comment (ryantwolf) [reply README.md:15]: Let's keep the diagram in the PR for now

File: README.md (hunk 1 above)
Comment (ryantwolf) [reply README.md]: @Maghoumi given you didn't change according to 2 & 3, I just want to confirm that the open source is needed for SEO and that "Large Language Models Data Curation" is correct (instead of "Large Language Model Data Curation")

**Review by ryantwolf — CHANGES_REQUESTED** [review]
Only a couple of nits this time. Thanks again!

File: README.md (hunk 6 above)
Comment (Maghoumi) [reply README.md:160]: Correct, the tutorial's overview says it's mean to "help" with data annotation and blending, which is why I thought to include that here.

I changed the wording so it only mentions annotation.

File: README.md (hunk 7 above)
Comment (Maghoumi) [reply README.md]: Ah I see, thanks for correcting my understanding. Revised.

File: README.md (hunk 2 above)
Comment (Maghoumi) [reply README.md:15]: Okay, kept.

**Comment by Maghoumi** [comment]
> @Maghoumi given you didn't change according to 2 & 3, I just want to confirm that the open source is needed for SEO and that "Large Language Models Data Curation" is correct (instead of "Large Language Model Data Curation")

@ryantwolf Oops, I missed your original comment. I changed it to "Large Language Model Data Curation". Yes the redundant open source emphasis is for SEO as there is no other mention of it in the readme file.

**Review by ryantwolf — APPROVED** [review]
Good with me. Thanks!
